python/cuml/manifold/t_sne.pyx

```diff
@@ -446,22 +447,24 @@ class TSNE(Base):
 
         if "handle" in state:
             del state["handle"]
-        self._assure_clean_memory()
+        self._assure_clean_memory(1)
         return state
 
     def __setstate__(self, state):
         super(TSNE, self).__init__(handle=None,
                                    verbose=(state['verbose'] != 0))
         self.__dict__.update(state)
-        self._assure_clean_memory()
+        self._assure_clean_memory(1)
         return state
 
-    def _assure_clean_memory(self):
+    def _assure_clean_memory(self, int collect=0):
         """
         TSNE is sensitive to what is currently in memory.
         Use Numba's garbabge collector to clean up already removed
         GPU memory.
         """
+        if collect == 1:
```

Comment:
Is this demonstrated to be make a difference? We shouldn't throw in GC's arbitrarily unless we can be convinced that it helps. Additionally, if you need a flag like this, it would be better to use a boolean True flag instead of 1

Reply:
Hmm I'll have to investigate more. However, I did add more checks in the actual C++ side. I *might* be actually accessing NULL memory, which might be the reason why it fails. Though not sure. I'll rerun all pytests in the cuml directory multiple many times to confirm.

Reply:
I changed it to a boolean flag. Anyways I am currently testing it on dgx to confirm if pytests succeed. I'll report my findings asap

Reply:
From the looks of things, in fact gc.collect() does not even have to be invoked and TSNE's test does not corrupt the other tests. I will monitor more though.